server/service/apple_mdm.go

```diff
@@ -2368,17 +2368,36 @@ func (svc *MDMAppleCheckinAndCommandService) DeclarativeManagement(*mdm.Request,
 // This method is executed after the request has been handled by nanomdm.
 //
 // [1]: https://developer.apple.com/documentation/devicemanagement/commands_and_queries
-func (svc *MDMAppleCheckinAndCommandService) CommandAndReportResults(r *mdm.Request, res *mdm.CommandResults) (*mdm.Command, error) {
-	// Sometimes we get results with Status == "Idle" which don't contain a command
-	// UUID and are not actionable anyways.
-	if res.CommandUUID == "" {
-		return nil, nil
+func (svc *MDMAppleCheckinAndCommandService) CommandAndReportResults(r *mdm.Request, cmdResult *mdm.CommandResults) (*mdm.Command, error) {
+	if cmdResult.Status == "Idle" {
+		// macOS hosts are considered unlocked if they are online any time
+		// after they have been unlocked. If the host has been seen after a
+		// successful unlock, take the opportunity and update the value in the
+		// db as well.
+		//
+		// TODO: sanity check if this approach is still valid after we implement wipe
+		host, err := svc.ds.HostByIdentifier(r.Context, cmdResult.UDID)
+		if err != nil {
+			return nil, ctxerr.Wrap(r.Context, err, "finding host by uuid")
+		}
+
+		// refetch mdmActions with the updated values.
+		mdmActions, err := svc.ds.GetHostLockWipeStatus(r.Context, host.ID, host.FleetPlatform())
+		if err != nil {
+			return nil, ctxerr.Wrap(r.Context, err, "get host mdm lock/wipe status")
+		}
+
+		if mdmActions.IsPendingUnlock() {
+			if err := svc.ds.CleanMacOSMDMLock(r.Context, host.ID); err != nil {
+				return nil, ctxerr.Wrap(r.Context, err, "cleaning macOS host lock/wipe status")
+			}
+		}
```

Comment:
@mna suggested we could run a query like this when we set host_seen_times:

```sql
UPDATE host_mdm_actions SET unlock_ref = NULL, lock_ref = NULL, unlock_pin = NULL WHERE host_id IN (?) AND unlock_ref IS NOT NULL AND unlock_pin IS NOT NULL
```

we should probably do the same here instead of the three different `ds` calls